backend/api/user_communities.go

```diff
@@ -0,0 +1,25 @@
+package api
+
+import (
+	"database/sql"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/git-adithyanair/cs130-group-project/token"
+)
+
+func (server *Server) GetUserCommunities(ctx *gin.Context) {
+
+	authPayload := ctx.MustGet(authorizationPayloadKey).(*token.Payload)
+
+	communities, err := server.queries.GetUserCommunities(ctx, authPayload.UserID)
+	if err != nil {
+		if err != sql.ErrNoRows {
+			ctx.JSON(http.StatusInternalServerError, errorResponse(err))
+		}
+		return
```

Comment:
Why do we not have to give a response before the return if there is an error that isn't empty rows?

Reply:
if there is some error and it is not the ErrNoRows, then something went wrong and we need to return a 500. if the error was ErrNoRows then communities will just be an empty list and we send that in the 200 response

Reply:
wait but doesn't the code return nothing on line 20 if `err == sql.ErrNoRows`? If it returns without `ctx.JSON...` does that automatically mean its sending 200 with an empty list?

Reply:
ohh yes good point I didn't catch that